sale_commission_product_criteria/models/commission.py

```diff
@@ -0,0 +1,299 @@
+        items = (
+            self.env["commission.item"]
+            .with_context(active_test=False)
+            .search([("commission_id", "=", self.id)])
+        )
+        if items:
+            items.write({"active": True})
+        return res
+
+    @api.onchange("commission_type")
+    def onchange_commission_type(self):
+        # Prevent commission_type change in certain cases
+        self.check_type_change_allowed_sale()
+        self.check_type_change_allowed_moves()
+
+    def check_type_change_allowed_sale(self):
+        sola_ids = self.env["sale.order.line.agent"].search(
+            [("commission_id", "=", self._origin.id)]
+        )
+        done_so_ids = sola_ids.filtered(lambda x: x.object_id.state in ["done", "sale"])
+        if done_so_ids:
+            raise ValidationError(
+                _(
+                    "There is done Sale Orders with this commission. "
+                    "Commission type change is not allowed."
+                )
+            )
+
+    def check_type_change_allowed_moves(self):
+        aila_ids = self.env["account.invoice.line.agent"].search(
+            [("commission_id", "=", self._origin.id)]
+        )
+        done_move_ids = aila_ids.filtered(
+            lambda x: x.object_id.parent_state == "posted"
+        )
+        if done_move_ids:
+            raise ValidationError(
+                _(
+                    "There is posted Account Move Lines with this commission. "
+                    "Commission type change is not allowed."
+                )
+            )
+
+
+class CommissionItem(models.Model):
+    _name = "commission.item"
+    _description = "Commission Item"
+    _order = "sequence, applied_on, categ_id desc, id desc"
+
+    sequence = fields.Integer(default=10)
+    active = fields.Boolean(default=True)
+    commission_id = fields.Many2one(
+        "sale.commission",
+        string="Commission Type",
+        domain=[("commission_type", "=", "product")],
+        required=True,
+    )
+    use_pricelist = fields.Boolean()
+    pricelist_id = fields.Many2one("product.pricelist")
+    product_tmpl_id = fields.Many2one(
+        "product.template",
+        "Product",
+        ondelete="cascade",
+        check_company=True,
+        help="Specify a template if this rule only applies to one "
+        "product template. Keep empty otherwise.",
+    )
+    product_id = fields.Many2one(
+        "product.product",
+        "Product Variant",
+        ondelete="cascade",
+        check_company=True,
+        help="Specify a product if this rule only applies "
+        "to one product. Keep empty otherwise.",
+    )
+    categ_id = fields.Many2one(
+        "product.category",
+        "Product Category",
+        ondelete="cascade",
+        help="Specify a product category if this rule only applies to "
+        "products belonging to this category or its children categories. "
+        "Keep empty otherwise.",
+    )
+    based_on = fields.Selection(
```

Comment:
I don't get the need of this.

Reply:
https://github.com/OCA/commission/pull/421/files#diff-db23ccfdfe6db97ef62e546b8909cb7e32eeb9e56e8f3087d0670decc07f83a2R17
check here

Reply:
OK, seems a bit awkward, but not being more picky.